homeassistant/components/media_player/dlna_dmr.py

```diff
@@ -0,0 +1,909 @@
+                              self,
+                              func.__name__, service_type)
+                raise NotImplementedError()
+
+            state_var = service.state_variable(state_variable_name)
+            if not state_var:
+                _LOGGER.error('requires_state_variable(): '
+                              '%s.%s: no state_variable: %s.%s',
+                              self,
+                              func.__name__,
+                              service_type,
+                              state_variable_name)
+                raise NotImplementedError()
+            return func(self, state_var, *args, **kwargs)
+        return wrapper
+    return call_wrapper
+
+
+def start_notify_view(hass):
+    """Register notify view."""
+    hass_data = hass.data[__name__]
+    name = 'notify_view'
+    if name in hass_data:
+        return hass_data[name]
+
+    view = UpnpNotifyView(hass)
+    hass_data[name] = view
+    hass.http.register_view(view)
+    return view
+
+
+def start_proxy_view(hass):
+    """Register proxy view."""
+    hass_data = hass.data[__name__]
+    name = 'proxy_view'
+    if name in hass_data:
+        return hass_data[name]
+
+    view = PickyDeviceProxyView(hass)
+    hass_data[name] = view
+    hass.http.register_view(view)
+    return view
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up DLNA DMR platform."""
+    if discovery_info and \
+       'upnp_device_type' in discovery_info and \
+       discovery_info['upnp_device_type'] != UPNP_DEVICE_MEDIA_RENDERER:
+        _LOGGER.debug('Device is not a MediaRenderer: %s',
+                      discovery_info.get('ssdp_description'))
+        return
+
+    is_picky = False
+    if config.get(CONF_URL) is not None:
+        url = config.get(CONF_URL)
+        name = config.get(CONF_NAME)
+    elif discovery_info is not None:
+        url = discovery_info['ssdp_description']
+        name = discovery_info['name']
+
+        # Samsung TVs are particular picky with regard to their sources
+        manufacturer = discovery_info.get('manufacturer', '').lower()
+        is_samsung = 'samsung' in manufacturer
+        is_tv = 'tv' in discovery_info.get('name', '').lower()
+        is_picky = is_samsung and is_tv
+
+    cfg_extra = {
+        CONF_MAX_VOLUME: config.get(CONF_MAX_VOLUME),
+        CONF_PICKY_DEVICE: config.get(CONF_PICKY_DEVICE) or is_picky,
+    }
+
+    # set up our Views, if not already done so
+    if __name__ not in hass.data:
+        hass.data[__name__] = {}
+
+    hass.async_run_job(start_notify_view, hass)
+    hass.async_run_job(start_proxy_view, hass)
+
+    from async_upnp_client import UpnpFactory
+    requester = HassUpnpRequester(hass)
+    factory = UpnpFactory(requester)
+    device = DlnaDmrDevice(hass, url, name, factory, **cfg_extra)
+
+    _LOGGER.debug("Adding device: %s", device)
+    add_devices([device])
+
+
+@asyncio.coroutine
```

Comment:
We have adopted PEP 492, please update accordingly, eg:

```
@asyncio.coroutine
def function():`
```
to 

```
async def function
```

`yield from` becomes `await`

Reply:
Will do.